rts-front/src/screens/register.rs

```diff
@@ -0,0 +1,144 @@
+use reqwasm::http::{Request, Response};
+use serde::{Deserialize, Serialize};
+use web_sys::HtmlInputElement;
+use yew::prelude::*;
+use yew_router::prelude::*;
+
+use crate::routes::Route;
+use crate::utils::alert_message;
+
+#[derive(Serialize)]
+struct RegistrationRequest {
+    email: String,
+    username: String,
+    password: String,
+}
+
+#[derive(Deserialize)]
+#[serde(untagged)]
+enum RegistrationResponse {
+    Successful,
+    Failed(String),
+}
+
+enum ResultOrResponse {
+    Result(RegistrationResponse),
+    Response(Response),
+}
+
+#[function_component(Registration)]
+pub fn registration_screen() -> Html {
+    let email = use_state(|| "".to_string());
```

Comment:
Can we consider using smart pointer on these ?

Reply:
Just like the login context, I don't think it's needed as use_state uses Rc internally: